variables.tf

```diff
@@ -110,9 +110,31 @@ variable "logging" {
   type = object({
     target_bucket = string
     target_prefix = string
+    target_object_key_format = optional(object({
+      format_type           = string                        # "simple" or "partitioned"
+      partition_date_source = optional(string, "EventTime") # Required if format_type is "partitioned"
+    }))
   })
-  default     = null
+
+  default = null
   description = "Logging configuration, logging is disabled by default."
+
+  validation {
```

Comment:
default = {
    target_bucket = "marwin"
    target_prefix = "log/"
    target_object_key_format = {
      format_type           = "partitioned"
    }
  }
  
  You validation has an bug, this scenario does not result in an error.
  
  Could you split up the validation? 1 valiation for format_type and then 1 validation for partitiona_data_source and then check if format_type is partitioned?
  
  Your current validation is very hard to understand

Reply:
This is correctly use. DeliveryTime is used as default when type = partioned. 
I added docmentation for the validation, it's complex but does exactly what it needs to do